assignment/assignment.hs

```diff
@@ -58,18 +88,22 @@ import Data.Char(intToDigit)
 -- write a data type named Measure that lets you specify a meansurement using
 -- either unit. Be sure to derive Eq and Show.
 
---data Measure = ...
+data Measure = MM Float | IN Float 
+    deriving (Show, Eq)
 
 -- Now write a function that takes a measurement. If it is already in millimeters,
 -- return it. Otherwise convert the measurement n inches to millimeters and
 -- return a new metric Measure value. (Use 25.4 as the conversion factor)
 
--- ensureMetric:: Measure -> Measure
+ensureMetric:: Measure -> Measure
+ensureMetric (MM measurement) = MM measurement
+ensureMetric (IN measurement) = MM (measurement*25.4)
 
 -- Finally use `map` to take a list of Measure values, and return a new
 -- list where all measurements are in metric.
 
--- listToMetric:: [ Measure ] -> [ Measure ]
+listToMetric measure = map ensureMetric measure
+
 
```

Comment:
Absolutely!

Only small suggestion would be to remove the `measure` parameter from both sides.